### PR TITLE
Modeling users

### PR DESCRIPTION
**Pull request by l-palermo**
Hi guys

Rspec: some of the tests are hidden because recall further functionalities such as update and delete a user. 
Validation of email and password has been provided.
The confirmation message of successful signup is anonymous due to sessions controller missing.

Please have a look at the flow checking out this branch and running the server.

Thanks

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  
+  validates :name,        presence: true
```
Comment (ajosephides) [inline app/models/user.rb]: This did not error / trigger a message for me (blank name field)

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  validates :email,       presence: true, format: { with:VALID_EMAIL_REGEX }
```
Comment (ajosephides) [inline app/models/user.rb]: This is cool and I see if working on page.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  validates :password,    presence: true, length: { maximum: 10, minimum: 6 }
```
Comment (ajosephides) [inline app/models/user.rb]: I did not get any sort of validation warning / message from the password field. As a user I didn't know it had to be between 6 and 10.

File: app/models/user.rb (hunk 2 above)
Comment (ajosephides) [reply app/models/user.rb]: I can sign up with the same email address multiple times. Should this be the case or should they be unique?

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1,3 +1,6 @@
+
+<h3 Welcome you are successfully signed up </h3>
```
Comment (ajosephides) [inline app/views/posts/index.html.erb]: I never see this message on the posts page

File: config/routes.rb [hunk 5]
```diff
@@ -2,4 +2,7 @@
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
   resources :posts
+  # get  '/signup',  to: 'users#new'
+  resources :users
+  root 'users#new'
```
Comment (ajosephides) [inline config/routes.rb]: Does this line set the homepage root url to the content from /users/new?

File: spec/routing/users_routing_spec.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    it "routes to #index" do
+      expect(:get => "/users").to route_to("users#index")
+    end
+
+    it "routes to #new" do
+      expect(:get => "/users/new").to route_to("users#new")
+    end
+
+    it "routes to #show" do
```
Comment (ajosephides) [inline spec/routing/users_routing_spec.rb]: I see there is a test for this but I can't seem to run these routes in the production app.
Is this deliberate?

File: app/models/user.rb (hunk 3 above)
Comment (l-palermo) [reply app/models/user.rb]: sorted

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (l-palermo) [reply app/views/posts/index.html.erb]: you will now

File: config/routes.rb (hunk 5 above)
Comment (l-palermo) [reply config/routes.rb]: yes it does

File: spec/routing/users_routing_spec.rb (hunk 6 above)
Comment (l-palermo) [reply spec/routing/users_routing_spec.rb]: it might have been changed, please recheck

File: app/models/user.rb (hunk 1 above)
Comment (ajosephides) [reply app/models/user.rb]: Cool. I see this error now.

File: app/models/user.rb (hunk 3 above)
Comment (ajosephides) [reply app/models/user.rb]: Cool. I see this on page now.

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (ajosephides) [reply app/views/posts/index.html.erb]: Confirm that I can see this

File: config/routes.rb (hunk 5 above)
Comment (ajosephides) [reply config/routes.rb]: Cool

File: app/models/user.rb (hunk 2 above)
Comment (ajosephides) [reply app/models/user.rb]: I understand email address don't need to be unique.

File: spec/controllers/posts_controller_spec.rb [hunk 7]
```diff
@@ -1,28 +1,31 @@
 require 'rails_helper'
 
 RSpec.describe PostsController, type: :controller do
+
   describe "GET /new " do
     it "responds with 200" do
-      get :new
+      get :new, params: { get: { session: session[:user_id] = 2 } }
```
Comment (ajosephides) [inline spec/controllers/posts_controller_spec.rb]: Why is this user id of 2?

Comment (l-palermo) [reply spec/controllers/posts_controller_spec.rb]: it is just a mocked user id to make the test passing

File: spec/requests/users_spec.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe "Users", type: :request do
+  describe "GET homepage" do
+    xit "acebook homepage" do
```
Comment (ajosephides) [inline spec/requests/users_spec.rb]: This should be removed / if it is pending.
The sign in PR has the root url and this test working.
Might be worth removing this from here before merging into master.

Comment (l-palermo) [reply spec/requests/users_spec.rb]: resolved

File: spec/controllers/posts_controller_spec.rb (hunk 7 above)
Comment (l-palermo) [reply spec/controllers/posts_controller_spec.rb]: This has now changed, these tests are pending

**Pull request merged**